Lib/test/test_hashlib.py

```diff
@@ -1196,6 +1196,29 @@ def test_file_digest(self):
             with open(os_helper.TESTFN, "wb") as f:
                 hashlib.file_digest(f, "sha256")
 
+    @unittest.skipUnless(support.check_sanitizer(thread=True), "only meaningful on free-threading")
+    def test_gh_128657(self):
+        def test_hashlib_sha256():
+            hash_obj = hashlib.sha256()
+
+        def closure(barrier):
+            barrier.wait()
+            test_hashlib_sha256()
+
+        num_workers = 40
+        num_runs = 20
+
+        for i in range(num_runs):
+            barrier = threading.Barrier(num_workers)
+            thrds = []
+
+            for i in range(num_workers):
+                thrds.append(thrd := threading.Thread(target=closure, args=(barrier,)))
+                thrd.start()
+
+            for thrd in thrds:
+                thrd.join()
```

Comment:
```suggestion
                thread.start()

            for thread in threads:
                thread.join()
```

I would prefer full names. Or if you want short names, you can use `t` for the thread variable (but I would prefer to keep `threads`).